docs/recipes/ExpoRouter.md

```diff
@@ -0,0 +1,1806 @@
+        RightAccessory={PasswordRightAccessory}
+      />
+
+      <Button
+        testID="login-button"
+        tx="loginScreen.tapToSignIn"
+        style={$tapButton}
+        preset="reversed"
+        onPress={login}
+      />
+    </Screen>
+  );
+});
+
+const $screenContentContainer: ViewStyle = {
+  paddingVertical: spacing.xxl,
+  paddingHorizontal: spacing.lg,
+};
+
+const $signIn: TextStyle = {
+  marginBottom: spacing.sm,
+};
+
+const $enterDetails: TextStyle = {
+  marginBottom: spacing.lg,
+};
+
+const $hint: TextStyle = {
+  color: colors.tint,
+  marginBottom: spacing.md,
+};
+
+const $textField: ViewStyle = {
+  marginBottom: spacing.lg,
+};
+
+const $tapButton: ViewStyle = {
+  marginTop: spacing.xs,
+};
+```
+
+</details>
+
+If you're familiar with the Ignite boilerplate, this is the same authentication screen you are used to - the only difference here is some of the imports now from from `src/*` rather than the relative paths. So keep that in mind if you're upgrading an existing application.
+
+### src/app/(app)/index.tsx
+
+If the user is successfully authenticated, we'll show them the welcome screen. Can you guess what the route will be by looking at the directory structure?
+
+Just the root route! Think about it in terms of web URLs, if arriving at `http://localhost:8081/` (in this case of local development), we'd expect to see the welcome screen. However, if we're not authenticated, we'll be redirected to `/log-in` to ask the user to log in.
+
+This JSX will be the same exact contents from `WelcomeScreen.tsx` in the original Ignite boilerplate with the exception of some import paths (using the TS aliases) and a simple update to `goNext`.
+
+Since we'll no longer use the `navigation` prop, we utilize `expo-router`'s [Imperative navigation](https://docs.expo.dev/routing/navigating-pages/#imperative-navigation) to navigate to the component demo Showroom next. We're using `.replace` since we don't need to get back to this route. You can read more about [Navigating between pages](https://docs.expo.dev/routing/navigating-pages/) at Expo's documentation.
+
+<details>
+  <summary>src/app/(app)/index.tsx</summary>
+
+```tsx
+import { router } from "expo-router";
+import { observer } from "mobx-react-lite";
+import React from "react";
+import { Image, ImageStyle, TextStyle, View, ViewStyle } from "react-native";
+import { Button, Text } from "src/components";
+import { isRTL } from "src/i18n";
+import { useStores } from "src/models";
+import { colors, spacing } from "src/theme";
+import { useHeader } from "src/utils/useHeader";
+import { useSafeAreaInsetsStyle } from "src/utils/useSafeAreaInsetsStyle";
+
+const welcomeLogo = require("assets/images/logo.png");
+const welcomeFace = require("assets/images/welcome-face.png");
+
+export default observer(function WelcomeScreen() {
+  const {
+    authenticationStore: { logout },
+  } = useStores();
+
+  function goNext() {
+    router.replace("/showroom");
```

Comment:
This doesn't exist at this point.

Reply:
I think that's okay - the app builds and shows login & welcome screens successfully at this point, and I figured it was simpler to keep it included in the full copypasta for simplicity, rather than having them go back and make edits in the next immediate section.